src/realm/list.hpp

```diff
@@ -250,16 +363,37 @@ class LnkLst : public Lst<ObjKey>, public ObjList {
     void remove_target_row(size_t link_ndx);
     void remove_all_target_rows();
 
+    iterator begin() const noexcept
+    {
+        return iterator{this, 0};
+    }
+    iterator end() const noexcept
+    {
+        return iterator{this, size()};
+    }
+
+    const BPlusTree<ObjKey>& get_tree() const
+    {
+        return m_keys.get_tree();
+    }
+
+protected:
+    bool update_if_needed() const final
+    {
+        if (m_keys.update_if_needed()) {
+            update_unresolved(*m_keys.m_tree);
+            return true;
+        }
+        return false;
+    }
+
 private:
     friend class ConstTableView;
     friend class Query;
 
-    // Sorted set of indices containing unresolved links.
-    mutable std::vector<size_t> m_unresolved;
+    Lst<ObjKey> m_keys;
```

Comment:
Can you come up with some other name? Without seeing this declaration first, I thought m_keys was just a BPlusTree with the keys in it. Something like m_list_obj_key - or something shorter.

Reply:
Good point, it's a very overloaded name. I'll try to come up with a better one.